control-plane/rest/openapi-specs/v0_api_spec.yaml

```diff
@@ -2978,6 +3028,39 @@ components:
       required:
         - spec
         - state
+    Snapshot:
+      description: |-
+        Snapshot Information
+      type: object
+      properties:
+        uuid:
+          $ref: '#/components/schemas/SnapshotId'
+        name:
+          description: Name of the snapshot
+          type: string
+        size:
+          description: |-
+            Size in bytes taken up by the snapshot in the backend.
+            This size may grow larger as the original volume gets written to, up to the volume size.
```

Comment:
`This size may grow larger as the original volume gets written to`
This sounds like size of the snapshot being taken varies, which isn't true. I guess this intends to say that `subsequent snapshots keep getting bigger if the volume is being written to`

Reply:
Okay. How about.
`This size on subsequent snapshots may grow larger as the original volume gets written to, up to the volume size.`

Reply:
My suggestion:
`As the original volume keeps getting written to, the subsequent snapshots have a larger size compared to older point in time snapshots, maximum up to the volume size.`

Reply:
Okay. Will update to this then.

Reply:
Subsequent snapshot sizes do not change, it's always the same size, what changes is volume itself which becomes thin, and so as we write to the volume, the snapshot "allocation" is effectively growing in size.

I think we better have 2 different sizes, one is the thin snapshot size, and the other is the "allocated" size of the snapshot (not really of the snapshot but the extra allocation that it's done as result of writing to the volume). So initially "allocated" is 0, and it grows up to the volume size.

Reply:
Also, should we have a spec/state kind of object here, similar to most other resources?

Reply:
Have split it, as per discussed.